v2/api/v2/addresspool_webhook.go

```diff
@@ -15,45 +18,67 @@ import (
 func (r *AddressPool) SetupWebhookWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
 		For(r).
+		WithDefaulter(&AddressPoolCustomDefaulter{}).
+		WithValidator(&AddressPoolCustomValidator{}).
 		Complete()
 }
 
+// AddressPoolCustomDefaulter is an empty struct that implements webhook.CustomDefaulter
+type AddressPoolCustomDefaulter struct{}
+
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 
 //+kubebuilder:webhook:path=/mutate-coil-cybozu-com-v2-addresspool,mutating=true,failurePolicy=fail,sideEffects=None,groups=coil.cybozu.com,resources=addresspools,verbs=create,versions=v2,name=maddresspool.kb.io,admissionReviewVersions={v1,v1beta1}
 
-var _ webhook.Defaulter = &AddressPool{}
+var _ webhook.CustomDefaulter = &AddressPoolCustomDefaulter{}
 
 // Default implements webhook.Defaulter so a webhook will be registered for the type
-func (r *AddressPool) Default() {
-	controllerutil.AddFinalizer(r, constants.FinCoil)
+func (r *AddressPoolCustomDefaulter) Default(ctx context.Context, obj runtime.Object) error {
+	addressPool, ok := obj.(*AddressPool)
+	if !ok {
+		return fmt.Errorf("expected an AddressPool object but got %T", obj)
+	}
+
+	controllerutil.AddFinalizer(addressPool, constants.FinCoil)
+	return nil
 }
 
+// AddressPoolCustomValidator is an empty struct that implements webhook.CustomValidator
+type AddressPoolCustomValidator struct{}
+
 // +kubebuilder:webhook:path=/validate-coil-cybozu-com-v2-addresspool,mutating=false,failurePolicy=fail,sideEffects=None,groups=coil.cybozu.com,resources=addresspools,verbs=create;update,versions=v2,name=vaddresspool.kb.io,admissionReviewVersions={v1,v1beta1}
 
-var _ webhook.Validator = &AddressPool{}
+var _ webhook.CustomValidator = &AddressPoolCustomValidator{}
 
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
-func (r *AddressPool) ValidateCreate() (warnings admission.Warnings, err error) {
-	errs := r.Spec.validate()
+func (r *AddressPoolCustomValidator) ValidateCreate(ctx context.Context, obj runtime.Object) (warnings admission.Warnings, err error) {
+	addressPool, ok := obj.(*AddressPool)
+	if !ok {
+		return nil, fmt.Errorf("expected an AddressPool object but got %T", obj)
+	}
+	errs := addressPool.Spec.validate()
 	if len(errs) == 0 {
 		return nil, nil
 	}
```

Comment:
```suggestion
	if errs := addressPool.Spec.validate(); len(errs) == 0 {
		return nil, nil
	}
```
nit
This is a minor comment, but I would like to match when there is an error (i.e. `if len(errs) != 0`).